src/components/refactor/dialogs/voltage-level-creation/voltage-level-creation-dialog.js

```diff
@@ -0,0 +1,211 @@
+} from 'components/refactor/utils/field-constants';
+import yup from 'components/refactor/utils/yup-config';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { createVoltageLevel } from 'utils/rest-api';
+import ModificationDialog from 'components/refactor/dialogs/commons/modificationDialog';
+
+import VoltageLevelCreationForm from './voltage-level-creation-form';
+import {
+    controleBusBarSectionLink,
+    controleUniqueHorizontalVertical,
+    controleUniqueId,
+} from './voltage-level-creation-utils';
+
+/**
+ * Dialog to create a load in the network
+ * @param currentNode The node we are currently working on
+ * @param studyUuid the study we are currently working on
+ * @param editData the data to edit
+ * @param onCreateVoltageLevel to create voltage level from other forms,
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+
+const emptyFormData = {
+    [EQUIPMENT_ID]: '',
+    [EQUIPMENT_NAME]: '',
+    [NOMINAL_VOLTAGE]: '',
+    [SUBSTATION_ID]: '',
+    [BUS_BAR_SECTIONS]: [],
+    [BUS_BAR_CONNECTIONS]: [],
+};
+
+const schema = yup.object().shape({
+    [EQUIPMENT_ID]: yup.string().required(),
+    [EQUIPMENT_NAME]: yup.string(),
+    [NOMINAL_VOLTAGE]: yup.string().required(),
+    [SUBSTATION_ID]: yup.string().required(),
+    [BUS_BAR_SECTIONS]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [ID]: yup.string().required(),
+                [NAME]: yup.string(),
+                [HORIZONTAL_POSITION]: yup.number().min(0).required(),
+                [VERTICAL_POSITION]: yup.number().min(0).required(),
+            })
+        )
+        .min(1, 'EmptyList/bbs')
+        .test('unique-positions', (values) =>
+            controleUniqueHorizontalVertical(values, 'SameHorizAndVertPos')
+        )
+        .test('unique-ids', (values) =>
+            controleUniqueId(values, 'DuplicateId')
+        ),
+    [BUS_BAR_CONNECTIONS]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [FROM_BBS]: yup.string().nullable().required(),
+                [TO_BBS]: yup.string().nullable().required(),
+                [SWITCH_KIND]: yup.string().nullable().required(),
+            })
+        )
+        .test('bus-bar-section-link', (values) =>
+            controleBusBarSectionLink(values, 'DisconnectorBetweenSameBusbar')
+        ),
+});
+
+const VoltageLevelCreationDialog = ({
+    editData,
+    currentNode,
+    studyUuid,
+    onCreateVoltageLevel = createVoltageLevel,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode?.id;
+    const { snackError } = useSnackMessage();
+    const equipmentPath = 'voltage-levels';
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const {
+        reset,
+        formState: { errors },
```

Comment:
do we need "errors" here ?
this will cause the whole to rerender everytime the "errors" object changes